cli/cmd/profile.go

```diff
@@ -87,3 +126,129 @@ func renderProfileTemplate(config *templateConfig, w io.Writer) error {
+	if err != nil {
+		return fmt.Errorf("Error parsing yaml: %s", err)
+	}
+
+	swagger := spec.Swagger{}
+	err = swagger.UnmarshalJSON(json)
+	if err != nil {
+		return fmt.Errorf("Error parsing OpenAPI spec: %s", err)
+	}
+
+	profile := sp.ServiceProfile{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      fmt.Sprintf("%s.%s.svc.cluster.local", options.name, options.namespace),
+			Namespace: controlPlaneNamespace,
+		},
+		TypeMeta: meta_v1.TypeMeta{
+			APIVersion: "linkerd.io/v1alpha1",
+			Kind:       "ServiceProfile",
+		},
+	}
+
+	routes := make([]*sp.RouteSpec, 0)
+	for path, item := range swagger.Paths.Paths {
+		pathRegex := pathToRegex(path)
+		if item.Delete != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodDelete, item.Delete.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Get != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodGet, item.Get.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Head != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodHead, item.Head.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Options != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodOptions, item.Options.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Patch != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodPatch, item.Patch.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Post != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodPost, item.Post.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Put != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodPut, item.Put.Responses)
+			routes = append(routes, spec)
+		}
+	}
+
+	profile.Spec.Routes = routes
+	output, err := yaml.Marshal(profile)
+	if err != nil {
+		return fmt.Errorf("Error writing Service Profile: %s", err)
+	}
+	w.Write(output)
+
+	return nil
+}
+
+func mkRouteSpec(path, pathRegex string, method string, responses *spec.Responses) *sp.RouteSpec {
+	return &sp.RouteSpec{
+		Name:            fmt.Sprintf("%s %s", method, path),
+		Condition:       toReqMatch(pathRegex, method),
+		ResponseClasses: toRspClasses(responses),
+	}
+}
+
+func pathToRegex(path string) string {
+	escaped := regexp.QuoteMeta(path)
+	replaced := pathParamRegex.ReplaceAllLiteralString(escaped, "[^/]*")
+	return fmt.Sprintf("^%s$", replaced)
+}
+
+func toReqMatch(path string, method string) *sp.RequestMatch {
+	return &sp.RequestMatch{
+		Path:   path,
+		Method: method,
+	}
+}
+
+func toRspClasses(responses *spec.Responses) []*sp.ResponseClass {
+	if responses == nil {
+		return nil
+	}
+	classes := make([]*sp.ResponseClass, 0)
+	for status := range responses.StatusCodeResponses {
+		cond := &sp.ResponseMatch{
+			Status: &sp.Range{
+				Min: uint32(status),
+				Max: uint32(status),
+			},
+		}
+		failure := false
+		if status >= 500 {
+			failure = true
+		}
+		classes = append(classes, &sp.ResponseClass{
+			Condition: cond,
+			IsFailure: !failure,
+		})
+	}
+	return classes
```

Comment:
similarly for output stability:
```go
sort.Slice(classes, func(i, j int) bool {
	return classes[i].Condition.Status.Min < classes[j].Condition.Status.Min
})
```
we could maybe go a bit farther than this and sort on `IsFailure` and `Max`, but this worked with my limited testing:
```bash
diff <(bin/go-run cli profile --open-api <(curl https://petstore.swagger.io/v2/swagger.json) pet) <(bin/go-run cli profile --open-api <(curl https://petstore.swagger.io/v2/swagger.json) pet)
```

Reply:
I wend with sorting the status keys prior to constructing response classes